src/GameLogic/TurnRunner.ts

```diff
@@ -27,9 +28,27 @@ export class TurnRunner {
 
 		level.entities.forEach(this.beforeUpdate);
 
-		level.entities.getEntitiesOfType(EntityType.Protagonist).forEach(entity => entity.update(turnState));
+		const protagonists = level.entities.getEntitiesOfType(EntityType.Protagonist);
```

Comment:
Now that we're switching to this verbose logic, let's move each entity type's runner into its own class, `ProtagonistUpdater` or something.